lib/Module/Starter/Smart.pm

```diff
@@ -170,11 +170,12 @@ sub create_t {
 }
 
 sub _create_t {
-    my $self = shift;
+    my $self     = shift;
+    my $testdir  = @_ == 2 ? 't' : shift;
```

Comment:
Cool now I follow you.  The change in the protocol is quite screwed up actually - I would have put `$directory` at the end rather than stuck in the middle.  Anyway it's not our problem after all.

Thanks for clarifying all these.  I'll do some sanity check and maybe put in some comments, then post it up to PAUSE.